.github/workflows/pull-request.yml

```diff
@@ -432,11 +433,72 @@ jobs:
           echo "Error: Close this PR and try again."
           exit 1
 
+  generate-docs:
+    name: Generate Auto Documentation
+    runs-on: ubuntu-latest
+    needs: [Code-Quality-Checks]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '22'
+
+      - name: Generate Documentation
+        run: |
+          # Install required dependencies
+          npm install -g typedoc
+          # Create auto-docs directory if it doesn't exist
+          mkdir -p docs/docs/auto-docs
+          # Generate documentation using TypeDoc
+          typedoc \
+            --out docs/docs/auto-docs \
+            --plugin typedoc-plugin-markdown \
+            --entryPoints "src/**/*.ts" \
+            --entryPoints "src/**/*.tsx" \
+            --exclude "**/*.test.ts" \
+            --exclude "**/*.test.tsx" \
+            --exclude "**/*.spec.ts" \
+            --exclude "**/*.spec.tsx"
+
+      - name: Setup Git Config
+        run: |
+          if ! git config user.name "GitHub Actions Bot" || \
+             ! git config user.email "41898282+github-actions[bot]@users.noreply.github.com"; then
+            echo "Failed to configure Git"
+            exit 1
+          fi
+
+      - name: Commit changes
+        env:
+          HEAD_REF: ${{ github.head_ref }}
+        run: |
+          # Stage the auto-generated docs
+          git add docs/docs/auto-docs
+          # Only commit if there are changes
+          if git diff --staged --quiet; then
+            echo "No changes to commit"
+          else
+            echo "Committing changes to documentation"
+            git commit -m "chore(docs): update auto-generated API documentation [skip ci]"
+            # Push to the feature branch
+             git push origin ${{ github.head_ref }}
+          fi
+
   Validate-Coderabbit:
     name: Validate CodeRabbit Approval
     runs-on: ubuntu-latest
     if: github.actor != 'dependabot[bot]'
-    needs: [Code-Quality-Checks, Test-Application, Start-App-Without-Docker, Docker-Start-Check]
+    needs:
+      [
```

Comment:
The opening bracket must be like this:

`needs: [`

What you have is invalid in YAML